waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/client/TunnelConfiguration.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (C) 2016-2017 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.waggledance.client;
+
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.jcraft.jsch.JSchException;
+import com.pastdev.jsch.tunnel.TunnelConnectionManager;
+
+import com.hotels.bdp.waggledance.api.WaggleDanceException;
+
+public class TunnelConfiguration {
+
+  private static final Logger LOG = LoggerFactory.getLogger(TunnelConfiguration.class);
+
+  private final HiveConf hiveConf;
+  private final TunnelConnectionManager tunnelConnectionManager;
+  private final String sshRoute;
+  private final String localHost;
+  private final String remoteHost;
+  private final int remotePort;
+
+  public TunnelConfiguration(
+      HiveConf hiveConf,
+      TunnelConnectionManager tunnelConnectionManager,
+      String sshRoute,
+      String localHost,
+      String remoteHost,
+      int remotePort) {
+    this.hiveConf = hiveConf;
+    this.tunnelConnectionManager = tunnelConnectionManager;
+    this.sshRoute = sshRoute;
+    this.localHost = localHost;
+    this.remoteHost = remoteHost;
+    this.remotePort = remotePort;
+  }
+
+  public HiveConf getConf() {
```

Comment:
This method isn't just getting a conf object, it's also creating a tunnel so the name is misleading. The name of the class similarly so. I'd expect a "XConfig" class to only hold config and then some other class would create a Tunnel (like a tunnel factory or some such and we already have a few factories here). Perhaps we need a whiteboard session to go over this proposed refactor?